modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -483,19 +469,88 @@ def process_circuits(
 
                 # extract job ID from response
                 jobid = cast(str, jobdict["job"])
-                if i == 0 and "batch_id" not in kwargs:
-                    # `batch-exec` field set to max batch cost for first job of batch
-                    # and to the id of first job of batch otherwise
-                    _ = self._api_handler.retrieve_job_status(
-                        jobid, use_websocket=kwargs.get("use_websocket", True)
-                    )
-                    batch_exec = jobid
+
                 handle = ResultHandle(jobid, json.dumps(ppcirc_rep))
                 handle_list.append(handle)
                 self._cache[handle] = dict()
 
         return handle_list
 
+    def _check_batchable(self) -> None:
+        if self.backend_info:
+            if not self.backend_info.misc.get("batching", False):
+                raise BatchingUnsupported()
+
+    def start_batch(
+        self,
+        max_batch_cost: int,
```

Comment:
note max_batch_cost is now a required parameter to reflect some upcoming changes in the API (different organisations having different maxima)